com.unity.netcode.gameobjects/Tests/Runtime/Metrics/MetricsDispatchTests.cs

```diff
@@ -0,0 +1,122 @@
+#if MULTIPLAYER_TOOLS
+using System;
+using System.Collections;
+using NUnit.Framework;
+using Unity.Multiplayer.Tools.NetStats;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Unity.Netcode.RuntimeTests.Metrics
+{
+    public class MetricsDispatchTests
+    {
+        private class MockMetricsObserver : IMetricObserver
+        {
+            public Action OnObserve;
+            public void Observe(MetricCollection collection) => OnObserve?.Invoke();
+        }
+
+        private int m_NumTicks;
+        private int m_NumDispatches;
+
+        private NetworkManager m_NetworkManager;
+        private NetworkTimeSystem timeSystem => m_NetworkManager.NetworkTimeSystem;
+        private NetworkTickSystem tickSystem => m_NetworkManager.NetworkTickSystem;
+        private uint m_TickRate = 1;
+
+        [SetUp]
+        public void SetUp()
+        {
+            Assert.IsTrue(NetworkManagerHelper.StartNetworkManager(
+                out m_NetworkManager,
+                NetworkManagerHelper.NetworkManagerOperatingMode.Host,
+                new NetworkConfig()
+            {
+                TickRate = m_TickRate
+            }));
+
+            InitNetworkManager();
+
+            var networkMetrics = m_NetworkManager.NetworkMetrics as NetworkMetrics;
+            networkMetrics.Dispatcher.RegisterObserver(new MockMetricsObserver()
+            {
+                OnObserve = ()=> m_NumDispatches++
```

Comment:
This is only ever overridden here, could this not just be the implementation of Observe() in the MockMetricsObserver?

Reply:
Then the mock metrics observer would need a reference back to the test fixture in order to update the value, which would be about the same amount of boilerplate IMO